File: Backend/examserver/src/main/java/com/exam/services/QuizService.java
package com.exam.services;

import java.util.List;
import java.util.Set;

import com.exam.models.exam.Category;
import com.exam.models.exam.Quiz;


public interface QuizService {
	
	public Quiz addQuiz(Quiz quiz);
	
	public Quiz updateQuiz(Quiz quiz);
	
	public Set<Quiz> getAllQuizzes();
	
	public Quiz getQuiz(Long quizId);
	
	public void deleteQuiz(Long quizId);
	
	public List<Quiz> getActiveQuiz();
	
	public List<Quiz> getActiveQuizOfCategory(Category category);

}

File: Frontend/exam-portal/src/app/Services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http: HttpClient) { }

  public getAllQuizzes(){
    return this.http.get(`${baseUrl}quiz/`)
  }
  public getQuiz(quizId : any){
    return this.http.get(`${baseUrl}quiz/${quizId}`)
  }

  public addQuiz(quiz : any){
    return this.http.post(`${baseUrl}quiz/add`, quiz)
  }

  public updateQuiz(quiz : any){
    return this.http.put(`${baseUrl}quiz/update`, quiz)
  }

  public deleteQuiz(quizId : any){
    return this.http.delete(`${baseUrl}quiz/${quizId}`)
  }

  public getQuizzesByCategory(cid : any){
    return this.http.get(`${baseUrl}quiz/category/all/${cid}`)
  }

  public getActiveQuizzesByCategory(cid : any){
    return this.http.get(`${baseUrl}quiz/category/active/${cid}`)
  }

  public getActiveQuiz(){
    return this.http.get(`${baseUrl}quiz/active`)
  }


}
